onnxruntime/core/providers/xnnpack/math/gemm.cc

```diff
@@ -0,0 +1,204 @@
+// Copyright(c) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT License.
+
+#include "gemm.h"
+#include "core/framework/transpose_helper.h"
+#include "core/providers/utils.h"
+
+namespace onnxruntime {
+namespace xnnpack {
+
+// Todo -
+// 1. Integrate activation layers - Cliping & Relu 
+// 2. Enable C matrix broadcasting - reuse "GemmBroadcastBias" function / logic
+// 3. Enable Quant ops
+// 4. Review possible consolidation of MatMul & Gemm 
+//
+
+bool Gemm::IsGemmOnnxNodeSupported(const NodeUnit& node_unit, const GraphViewer& graph) {
+  bool supported = false;
+  const onnxruntime::Node& node = node_unit.GetNode();
+
+  // use do {} while(false) so it's easier to set a breakpoint on the return
+  do {
+    ConstPointerContainer<std::vector<NodeArg*>> input_defs = node.InputDefs();
+
+    const auto alpha = node.GetAttributes().find("alpha");
+    if ((*alpha).second.f() != 1.0) break;
+
+    const auto beta = node.GetAttributes().find("beta");
+    if ((*beta).second.f() != 1.0) break;
+
+
+    const NodeArg* A_arg = input_defs[0];
+    const NodeArg* B_arg = input_defs[1];
+    const NodeArg* C_arg = input_defs.size() == 2 ? nullptr : input_defs[2];
+
+    // we only support float currently
+    const auto* A_type = A_arg->TypeAsProto();
+
+    if (A_type == nullptr || 
+        A_type->tensor_type().elem_type() != ONNX_NAMESPACE::TensorProto_DataType_FLOAT) {
+      break;
+    }
+
+    // B & C matrices must be constant
+    if (!graph.IsConstantInitializer(B_arg->Name(), true)) {
+      break;
+    }
+
+    if (input_defs.size() == 3 && !graph.IsConstantInitializer(C_arg->Name(), true)) {
+      break;
```

Comment:
nit: still need to check Exists. The NodeArg might still be present with an empty name.

`if (C_arg && C_arg->Exists() && ...`